File: platform/lib/domain/src/main/kotlin/io/hamal/lib/domain/request/Topic.kt
package io.hamal.lib.domain.request

import io.hamal.lib.domain._enum.TopicType
import io.hamal.lib.domain.vo.*


interface TopicCreateRequest {
    val name: TopicName
    val type: TopicType
}

data class TopicCreateRequested(
    override val requestId: RequestId,
    override val requestedBy: AuthId,
    override var requestStatus: RequestStatus,
    val id: TopicId,
    val logTopicId: LogTopicId,
    val workspaceId: WorkspaceId,
    val namespaceId: NamespaceId,
    val name: TopicName,
    val type: TopicType
) : Requested()


interface TopicAppendEntryRequest {
    val topicId: TopicId
    val payload: TopicEventPayload
}

data class TopicAppendEventRequested(
    override val requestId: RequestId,
    override val requestedBy: AuthId,
    override var requestStatus: RequestStatus,
    val id: TopicId,
    val payload: TopicEventPayload
) : Requested()

File: app/fn-guru/src/main/frontend/src/types/topic.ts
export interface TopicCreateRequested {
    requestId: string;
    requestStatus: string;
    id: string;
    workspaceId: string;
    namespaceId: string;
    type: string;
}

export type TopicType = 'Namespace' | 'Workspace' | 'Public'

export interface Topic {
    id: string;
    name: string;
    type: TopicType;
}

export interface TopicList {
    topics: Array<TopicListItem>;
}

export interface TopicListItem {
    id: string;
    name: string;
    type: TopicType;
}

export interface TopicEventAppendRequested {
    requestId: string;
    requestStatus: string;
    id: string;
}

export interface TopicEventPayload {
    payload: string
}